docs/enhancements/arm64-cpu-architecture.md

```diff
@@ -0,0 +1,192 @@
+---
+title: arm64-cpu-architecture
+authors:
+  - "@danielerez"
+creation-date: 2021-07-29
+last-updated: 2021-07-29
+---
+
+# ARM64 CPU architecture support
+
+## Summary
+
+The default host CPU architecture supported for installation is x86_64 in Assisted Installer. In order to support additional CPU architectures, the assisted-service should accept and handle selection of arm64 CPU architecture as well.
+
+## Motivation
+
+Although x86_64 is popular in many use cases, other CPU architectures are common in other segments.  For example, ARM is popular in edge use cases.  OpenShift has recently added support for arm64 and it is important to add this support to Assisted Installer.
+
+### Goals
+
+* Allow installation of clusters that are running on arm64 hosts.
+* The introduced infrastructure should facilitate future support of more architectures (e.g. PPC).
+
+### Non-Goals
+
+* The assisted-service and assisted-image-service will not be built for arm64.
+* No support for clusters with hosts of multiple architectures (e.g. x86_64 + arm64).
+* Implementation for kube-api as ClusterImageSet and AgentServiceConfig CRDs. Though the feature design should consider implications.
+* Day2 and Late binding won't be supported in first phase.
+* On-Prem flows, for now the focus is for SaaS users.
+* No baremetal platform support for arm64 in 4.9. I.e. only SNO for now.
+
+## Proposal
+
+### Host images
+
+The following images should also be built for arm64:
+  * assisted-installer
+  * assisted-installer-controller
+  * assisted-installer-agent
+
+This should be handled in similar to the current flows by modifying relevant Dockerfiles to support arm64 build. The images should be published to the same registries and follow same naming conventions. I.e. The best approach would be using the same image URI for any architecture. Which means, minimal changes in the assisted-service, as the architecture specific image would be fetched according to the machine pulling the image.
+
+### OCP versions
+
+[Versions](https://github.com/openshift/assisted-service/blob/master/internal/versions/versions.go) should
+have CPU architecture granularity for each API. So that OCP release images and RHCOS images of multiple architectures could be supported.
+
+The assisted-service should support the arm64 variant of RHCOS images and OCP release image (for OpenShift >= 4.9).
+
+As a preparation for supporting multiple CPU architectures in Version, the RHCOS images information from OPENSHIFT_VERSIONS environment variable should be extracted into a new variable (e.g. RHCOS_VERSIONS).
+We should probably use a simple array structure instead of mapping, to align with the [AgentServiceConfig](https://github.com/openshift/assisted-service/blob/77cb9d3348dafd2abd61fd7c8066e7e7a7805d95/config/samples/agent-install.openshift.io_v1beta1_agentserviceconfig.yaml#L18-L22) structure.
```

Comment:
I'm not sure this is an important consideration. I'd rather the structure @mhrivnak suggested in https://github.com/openshift/assisted-service/pull/2305#discussion_r681169284.

Reply:
Good point, perhaps we should just keep using the current structure, for example as already used in https://github.com/openshift/assisted-image-service#configuration (RHCOS_VERSIONS).
Since the operator have to parse the json object any way (and only once), maybe it's better to keep the mapping?
@flaper87 @mhrivnak wdyt?

Reply:
So we'll try to align with the k8s API convention if not too cumbersome: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps